src/addBusinessDays/index.ts

```diff
@@ -37,110 +37,65 @@ export default function addBusinessDays(
   requiredArgs(2, arguments)
   const options = dirtyOptions || {}
   const exceptions = options.exceptions || {}
-  const businessDays =
-    options.businessDays == null
-      ? [1, 2, 3, 4, 5]
-      : options.businessDays.filter((number) => number < 7).map(toInteger)
+  const businessDays = options.businessDays ?? [1, 2, 3, 4, 5]
 
-  const date = toDate(dirtyDate)
-  const amount = toInteger(dirtyAmount)
-  let accountedForExceptions = 0
-  const isExcepted = (date: Date): boolean | null => {
-    if (options.exceptions) {
-      const exception =
-        Object.keys(options.exceptions).find((e) =>
-          isSameDay(new Date(e), date)
-        ) || ''
-      if (options.exceptions[exception] !== undefined) {
-        accountedForExceptions += 1
-      }
-      return options.exceptions[exception]
-    }
-    return null
+  // Throw a RangeError if businessDays includes a number greater than 6
+  if (businessDays?.filter((number) => number > 6).length > 0) {
+    throw new RangeError('business days must be between 0 and 6')
   }
-  const isNonWorkingDay = (date: Date) =>
-    isExcepted(date) === false || !businessDays.includes(date.getDay())
+
+  const initialDate = toDate(dirtyDate)
+  const amount = toInteger(dirtyAmount)
 
   if (isNaN(amount)) return new Date(NaN)
 
-  if (date.toString() === 'Invalid Date') {
-    return date
+  if (initialDate.toString() === 'Invalid Date') {
+    return initialDate
   }
-  const startedOnNonWorkingDay = isNonWorkingDay(date)
-  const hours = date.getHours()
-  const sign = amount < 0 ? -1 : 1
-  const fullWeeks = toInteger(amount / businessDays.length)
-  date.setDate(date.getDate() + fullWeeks * 7)
 
-  // Get remaining days not part of a full week
-  let restDays = Math.abs(amount % businessDays.length)
-  // Loops over remaining days
-  while (restDays > 0) {
-    date.setDate(date.getDate() + sign)
-    if (!isNonWorkingDay(date)) restDays -= 1
+  // returns a boolean if the date has an exception
+  // true means the date is a working day
+  // false means the date is not a working day
+  const findException = (date: Date): boolean | undefined => {
+    return exceptions[format(date, 'MM/dd/yy')]
```

Comment:
@chr1sjf0x This now formats the date to avoid doing a `find`. However, this check won't work correctly if the exception is not in the right format. 

We could loop through each exception to check for whether the exception has the correct format, but would that somewhat defeat the purpose of this change? Other options that I'm not thinking of?

Reply:
I'd be inclined to say having them in the correct format is a requirement for this to work correctly which we don't/shouldn't need to validate at this level. It is stated in the javadoc associated with the exceptions field after all 🤷